httpclient/httpclient.go

```diff
@@ -26,3 +29,30 @@ func CloneHTTPHeader(in http.Header) http.Header {
 	}
 	return out
 }
+
+// MustHTTPClient wraps delegate transports with logging, metrics, rate limiting, retryable, user agent, request id
+// and panics if any error occurs.
+func MustHTTPClient(cfg *Config, userAgent, reqType string, delegate http.RoundTripper) *http.Client {
+	if delegate == nil {
+		delegate = http.DefaultTransport
+	}
+
+	delegate = NewLoggingRoundTripper(delegate, reqType)
+	delegate = NewMetricsRoundTripper(delegate, reqType)
+
+	delegate, err := NewRateLimitingRoundTripperWithOpts(
+		delegate, cfg.RateLimits.Limit, cfg.RateLimits.TransportOpts(),
```

Comment:
It would be a good idea to accept `LoggerProvider func(ctx context.Context) log.FieldLogger` as an optional parameter in the constructors (`NewHTTPClient` and `MustNewHTTPClient`). This provider can then be passed as an option to `NewRateLimitingRoundTripperWithOpts`.

By default, the `LoggerProvider` should be set to `middleware.GetLoggerFromContext`.

Additionally, this `LoggerProvider` should also be utilized for the `LoggingRoundTripper`.